flask_admin/model/widgets.py

```diff
@@ -61,3 +61,112 @@ def __call__(self, field, **kwargs):
         kwargs.setdefault('data-placeholder', placeholder)
 
         return HTMLString('<input %s>' % html_params(name=field.name, **kwargs))
+
+
+class XEditableWidget(object):
+    """
+        WTForms widget that provides in-line editing for the list view.
+
+        Determines how to display the x-editable/ajax form based on the
+        field inside of the FieldList (StringField, IntegerField, etc).
+    """
+    def __call__(self, field, **kwargs):
+        value = kwargs.pop("value", "")
+
+        kwargs.setdefault('data-role', 'x-editable')
+        kwargs.setdefault('data-url', './ajax/update/')
+
+        kwargs.setdefault('id', field.id)
+        kwargs.setdefault('name', field.name)
+        kwargs.setdefault('href', '#')
+
+        if not kwargs.get('pk'):
+            raise Exception('pk required')
+        kwargs['data-pk'] = str(kwargs.pop("pk"))
+
+        kwargs['data-csrf'] = kwargs.pop("csrf", "")
+
+        # gets the first entry (subfield) from FieldList (field) and uses the
+        # subfield.type (`kwargs_` method) to determine kwargs
+        subfield = field.entries[0]
+        try:
+            kwargs = getattr(self, 'kwargs_' + subfield.type)(subfield, kwargs)
```

Comment:
I was thinking about method that'll accept subfield and return `kwargs`. Inside it can have bunch of ifs, call child methods like you did, etc. As long as this method is overridable, it should do the trick.